indico/modules/search/controllers.py

```diff
@@ -15,50 +18,60 @@
 from indico.modules.events import Event
 from indico.modules.groups import GroupProxy
 from indico.modules.search.base import IndicoSearchProvider, SearchTarget, get_search_provider
-from indico.modules.search.schemas import CategoryResultSchema, EventResultSchema
+from indico.modules.search.schemas import DetailedCategorySchema, EventSchema
 from indico.modules.search.views import WPSearch
+from indico.util.caching import memoize_request
 from indico.web.args import use_kwargs
 from indico.web.rh import RH
 
 
+@memoize_request
+def get_groups(user):
+    access = [user.identifier] + [x.identifier for x in user.local_groups]
+    if user.can_get_all_multipass_groups:
+        access += [GroupProxy(x.name, x.provider.name).identifier
```

Comment:
```suggestion
        access += [GroupProxy(x.name, x.provider.name, x).identifier
```

While `.identifier` should never trigger a lookup of the remote group data, passing the group explicitly via the third constructor argument ensures that it's always available.

Reply:
Sounds good